src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -587,7 +652,10 @@ class Namer { typer: Typer =>
      *  to pick up the context at the point where the completer was created.
      */
     def completeInCreationContext(denot: SymDenotation): Unit = {
-      addAnnotations(denot)
+      original match {
+        case original: MemberDef => addAnnotations(denot.symbol, original)
+        case _ =>
+      }
```

Comment:
Annotations, and classes or objects containing them, might be completed out of order as part of the annotations phase. It seems necessary that these symbols get annotated on completion, or otherwise we'd introduce a compilation order dependency.